src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -128,9 +160,17 @@ public static bool IsDefinedInMongoLinqOrSystemLinq(this ISymbol symbol)
         typeSymbol.TypeKind == TypeKind.Class &&
         !typeSymbol.IsAnonymousType;
 
+    public static bool IsSupportedBsonAttribute(this ITypeSymbol typeSymbol) =>
+        typeSymbol?.ContainingNamespace?.ToDisplayString() == NamespaceMongoDBBsonAttributes &&
+        s_supportedBsonAttributes.Contains(typeSymbol?.Name);
```

Comment:
Lets switch checks order, to avoid invoking `ToDisplayString` if not needed (seems more heavy). Also in `IsSupportedSystemType`

Reply:
Done